### PR TITLE
feat: Add non alphanumeric removal for better url and filename support

### PR DESCRIPTION
**Pull request by RubenMateus**

### CODE REVIEW

File: Sluggy/Sluggy.cs [hunk 1]
```diff
@@ -70,10 +72,11 @@ public static string ToSlug(this string text, string separator, ITranslationStra
...
 
             return text
-                .Split()
-                .Where(t => t.Length != 0)
-                .Select(t => strategy.Translate(t))
-                .Join(separator);
+                 .Split()
+                 .Where(t => t.Length != 0)
+                 .Select(t => strategy.Translate(t))
+                 .Where(t => t.Length != 0)
```
Comment (sj-freitas) [inline Sluggy/Sluggy.cs:78]: It seems strange to have this 2nd check. Shouldn't the translate strategies already ensure this step isn't needed?

Comment (sj-freitas) [reply Sluggy/Sluggy.cs:78]: Or maybe remove the first Where.

File: Sluggy.Tests/Strategies/NonAlphaNumericStrategyTests.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+using Xunit;
+
+namespace Sluggy.Tests
+{
+    public class NonAlphaNumericStrategyTests
+    {
+        [Trait("Project", "Sluggy")]
+        [Theory(DisplayName = "Should remove nonalphanumeric characters from string")]
+        [InlineData("<>*.,;´`'~^!#%$&/()=}{[]@£€§¨+|", "")]
+        [InlineData("a<>*.,;´`'~^!#%$&/()=}{[]@£€§¨+| -a", "a a")]
```
Comment (sj-freitas) [inline Sluggy.Tests/Strategies/NonAlphaNumericStrategyTests.cs:12]: Weird to have an example where the response has a space. Yes this is a correct behaviour but the strategy should never be computed with spaces (as it's split before)

Comment (RubenMateus) [reply Sluggy.Tests/Strategies/NonAlphaNumericStrategyTests.cs:12]: That makes sense, will fix the test.

File: Sluggy/Sluggy.cs (hunk 1 above)
Comment (RubenMateus) [reply Sluggy/Sluggy.cs:78]: yeah i know.
The first where is to avoid the strategies to process empty strings.
The second is to cleanup any empty strings.

**Pull request merged**